
training of backgammon
(1, 34)
Traceback (most recent call last):
  File "/root/src/pgx/pgx/workspace/backgammon_trainig/ppo.py", line 345, in <module>
    out = train(config, rng)
  File "/root/src/pgx/pgx/workspace/backgammon_trainig/ppo.py", line 277, in train
    network_params = network.init(_rng, init_x)
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/linen/module.py", line 1640, in init
    _, v_out = self.init_with_output(
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/jax/_src/traceback_util.py", line 166, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/linen/module.py", line 1545, in init_with_output
    return init_with_output(
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/core/scope.py", line 965, in wrapper
    return apply(fn, mutable=mutable, flags=init_flags)({}, *args, rngs=rngs,
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/core/scope.py", line 933, in wrapper
    y = fn(root, *args, **kwargs)
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/linen/module.py", line 2121, in scope_fn
    return fn(module.clone(parent=scope), *args, **kwargs)
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/linen/module.py", line 432, in wrapped_module_method
    return self._call_wrapped_method(fun, args, kwargs)
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/linen/module.py", line 864, in _call_wrapped_method
    y = fun(self, *args, **kwargs)
  File "/root/src/pgx/pgx/workspace/backgammon_trainig/ppo.py", line 52, in __call__
    x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/linen/pooling.py", line 91, in avg_pool
    y = pool(inputs, 0., lax.add, window_shape, strides, padding)
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/linen/pooling.py", line 60, in pool
    assert inputs.ndim == len(dims), f"len({inputs.shape}) != len({dims})"
jax._src.traceback_util.UnfilteredStackTrace: AssertionError: len((1, 64)) != len((2, 2, 1))
The stack trace below excludes JAX-internal frames.
The preceding is the original exception that occurred, unmodified.
--------------------
The above exception was the direct cause of the following exception:
Traceback (most recent call last):
  File "/root/src/pgx/pgx/workspace/backgammon_trainig/ppo.py", line 345, in <module>
    out = train(config, rng)
  File "/root/src/pgx/pgx/workspace/backgammon_trainig/ppo.py", line 277, in train
    network_params = network.init(_rng, init_x)
  File "/root/src/pgx/pgx/workspace/backgammon_trainig/ppo.py", line 52, in __call__
    x = nn.avg_pool(x, window_shape=(2, 2), strides=(2, 2))
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/linen/pooling.py", line 91, in avg_pool
    y = pool(inputs, 0., lax.add, window_shape, strides, padding)
  File "/root/.pyenv/versions/3.10.0/lib/python3.10/site-packages/flax/linen/pooling.py", line 60, in pool
    assert inputs.ndim == len(dims), f"len({inputs.shape}) != len({dims})"
AssertionError: len((1, 64)) != len((2, 2, 1))